Tests/SDFCoreTests/MacroInitializerSorterTests.swift

```diff
@@ -1,201 +1,290 @@
+      try await #expect(sut.run(onContent: content.rawValue).contains(expected))
+    }
+
+    @Test(
+      "With a type that uses the macro inside of another type. It does not sort the outer init parameters",
+      arguments: TestData.WithNestedDefinitionMacroInside.allCases)
+    func run_doesNotSortOuterNested(
+      content: TestData.WithNestedDefinitionMacroInside
+    ) async throws {
+      let expected = "  init(b: String, a: Int) {}"
+      try await #expect(sut.run(onContent: content.rawValue).contains(expected))
+    }
 
-  @Test(
-    "With a type that uses the macro that defines a nested type with an unsorted initializer. It does not sort the nested init parameters",
-    arguments: TestData.WithNestedDefinitions.allCases)
-  func run_doesNotSortNested(content: TestData.WithNestedDefinitions) async throws {
-    let expected = """
-        init(d: String,
-             c: Int) {}
-    """
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
-  }
-  
-  @Test(
-    "With a type that uses the macro inside of another type. It sorts the inner init parameters",
-    arguments: TestData.WithNestedDefinitionMacroInside.allCases)
-  func run_doesSortInnerWhenNested(content: TestData.WithNestedDefinitionMacroInside) async throws {
-    let expected = "    init(c: Int, d: String) {}"
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
   }
-  
-  @Test(
-    "With a type that uses the macro inside of another type. It does not sort the outer init parameters",
-    arguments: TestData.WithNestedDefinitionMacroInside.allCases)
-  func run_doesNotSortOuterNested(content: TestData.WithNestedDefinitionMacroInside) async throws {
-    let expected = "  init(b: String, a: Int) {}"
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
+
+  struct AssignmentTests {
+    let sut = MacroInitializerSorter(macroName: "TestMacro")
+
+    @Test(
+      "With a self assignments in the intitializer. It sorts them",
+      arguments: TestData.types
+    )
+    func run_sortsAssignments(type: String) async throws {
+      let content = """
+        @TestMacro
+        \(type) Some {
+          init(a: Int, b: String) {
+            self.b = b
+            self.a = a
+          }
+        }
+        """
+      let expected = """
+            self.a = a
+            self.b = b
+        """
+      #expect(try await sut.run(onContent: content).contains(expected))
+    }
+
+    @Test(
+      "With self assignments and other values in the intitializer. It sorts them with all the self assignments on top",
+      arguments: TestData.types
+    )
+    func run_sortsAssignmentsWithOther(type: String) async throws {
+      let content = """
+        @TestMacro
+        \(type) Some {
+          init(a: Int) {
+            b = Type()
+            self.a = a
+          }
+        }
+        """
+      let expected = """
+            self.a = a
+            b = Type()
```

Comment:
it might be nicer to only sort `self.blah = blah` calls relative to one another, and **not** change their position relative to other kinds of assignments. That way if another assignment utilizes the value of a `self.` assignment, we aren't introducing a logic-breaking change via the linter

Reply:
Could you expand on this? If you could give me some test cases that'd be super useful! I'm not entirely sure what the expected behavior for moving the `self.` statements is when there are other statements in between.

Reply:
here's a contrived example:
```
super.init()
self.c = c
self.b = b
a = Self.createValue(from: self.b)
```

should become:
```
super.init()
self.b = b
self.c = c
a = Self.createValue(from: self.b)
```

Similarly (though this one doesn't break the logic if we fail this test)
```
self.c = c
self.b = b
a = A()
self.e = e
self.d = d
```

would become:
```
self.b = b
self.c = c
a = A()
self.d = d
self.e = e
```

Similarly:
```
self.c = c
self.b = b
print("hello there")
self.e = e
self.d = d
```

becomes:
```
self.b = b
self.c = c
print("hello there")
self.d = d
self.e = e
```